cryosparc/tools.py

```diff
@@ -143,19 +148,47 @@ def __init__(
         assert password, "Invalid or unspecified password"
 
         self.cli = CommandClient(
-            service="command_core", host=host, port=base_port + 2, headers={"License-ID": license}, timeout=timeout
+            service="command_core",
+            host=host,
+            port=base_port + 2,
+            headers={"License-ID": license},
+            timeout=timeout,
         )
         self.vis = CommandClient(
-            service="command_vis", host=host, port=base_port + 3, headers={"License-ID": license}, timeout=timeout
+            service="command_vis",
+            host=host,
+            port=base_port + 3,
+            headers={"License-ID": license},
+            timeout=timeout,
         )
         self.rtp = CommandClient(
-            service="command_rtp", host=host, port=base_port + 5, headers={"License-ID": license}, timeout=timeout
+            service="command_rtp",
+            host=host,
+            port=base_port + 5,
+            headers={"License-ID": license},
+            timeout=timeout,
         )
         try:
             self.user_id = self.cli.get_id_by_email_password(email, password)  # type: ignore
+            cs_version: str = self.cli.get_running_version()  # type: ignore
         except Exception as e:
             raise RuntimeError("Could not complete CryoSPARC authentication with given credentials") from e
 
+        if cs_version and VERSION_REGEX.match(cs_version):
+            minor_cs_version = ".".join(cs_version[1:].split(".")[:2])  # e.g., v4.1.0 -> 4.1
+            minor_tools_version = ".".join(__version__.split(".")[:2])  # e.g., 4.1.0 -> 4.1
+            tools_prerelease_url = "https://github.com/cryoem-uoft/cryosparc-tools/archive/refs/heads/develop.zip"
+            if minor_cs_version != minor_tools_version:
```

Comment:
should there be a check/message for major version mismatch as well?

Reply:
It's checking the first two version components, e.g., "vX.Y", not just the "Y". So a major mismatch will register. The variables could have a different to reflect that they are "Up to the minor version component", though I can't think of a good concise name for that 😅

Reply:
ah I see, maybe `major_minor_cs_version` could work. Not big deal though

Reply:
Done!